runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -323,8 +323,172 @@ J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+      cg->addSnippet(snippet);
+      }
+
+   // One of the object is array. Test equality of two objects' classes.
+#ifdef OMR_GC_COMPRESSED_POINTERS
+   // must read only 32 bits
+   generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp2Reg,
+         new (cg->trHeapMemory()) TR::MemoryReference(obj2Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+   generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+         new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+#else
+   generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp2Reg,
+         new (cg->trHeapMemory()) TR::MemoryReference(obj2Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+   generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp1Reg,
+         new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+#endif
+   TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, tmp2Reg);
+   TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, tmp1Reg);
+   generateCompareInstruction(cg, node, tmp1Reg, tmp2Reg);
+
+   // If either object is known to be of primitive component type,
+   // we are done: since both of them have to be of equal class.
+   if (node->isArrayChkPrimitiveArray1() || node->isArrayChkPrimitiveArray2())
+      {
+      if (NULL == snippetLabel)
+         {
+         snippetLabel = generateLabelSymbol(cg);
+         gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_NE);
+         snippet = new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference(), doneLabel);
+         cg->addSnippet(snippet);
+         }
+      else
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_NE);
+      }
+   // We have to take care of the un-equal class situation: both of them must be of reference array
+   else
+      {
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ);
+
+      // Object1 must be of reference component type, otherwise throw exception
+      if (!node->isArrayChkReferenceArray1())
+         {
+
+         // Loading the Class Pointer -> classDepthAndFlags
+#ifdef OMR_GC_COMPRESSED_POINTERS
+         generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+#else
+         generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+#endif
+         TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, tmp1Reg);
```

Comment:
Why load from VFT field of `obj1` and generate VFTMask again?  You can reuse the value in `tmp1Reg` at `generateCompareInstruction(cg, node, tmp1Reg, tmp2Reg)`.